### ISSUE TITLE
Memory leak for list and dict results

### ISSUE THREAD

**Issue by bobh66** [doloopwhile/pyjq#32]
@doloopwhile - there is a memory leak in jv_to_pyobj() when it returns list and dict results.

The problem is that the jv_array_get(), jv_object_iter_key() and jv_object_iter_value() methods all call jv_copy() and increment the reference count of the return value, so the results from these methods have to be jv_free()'d.

This sample code shows the memory leak in the list and dict case, and the lack of a memory leak in the string case:

```python
import os
import psutil
from pyjq import compile

recipe = "{x: .foo}"
c = compile(recipe)

process = psutil.Process(os.getpid())
before = process.memory_info().rss

for k in range(0, 10000):
    x = c.first({'foo': 'abcdefghijklmnopqrstuvwxyz'})

after = process.memory_info().rss

print(f"dict: {after - before}")

recipe = "[.foo]"
c = compile(recipe)

before = process.memory_info().rss

for k in range(0, 10000):
    x = c.first({'foo': 'abcdefghijklmnopqrstuvwxyz'})

after = process.memory_info().rss

print(f"list: {after - before}")

recipe = ".foo"
c = compile(recipe)

process = psutil.Process(os.getpid())
before = process.memory_info().rss

for k in range(0, 10000):
    x = c.first({'foo': 'abcdefghijklmnopqrstuvwxyz'})

after = process.memory_info().rss

print(f"string: {after - before}")
```
Note that the leak only seems to happen for large numbers of invocations of the first/all/one methods.  If the range maximum is 100 or 1000 it doesn't seem to leak, but 10000 shows the leak.

I have a patch that calls jv_free() on the list value, and the dict key/value, and the same sample script runs with no memory loss.  I'll submit a PR for the change.

Thanks

**Comment by bobh66** [comment]
@doloopwhile - Thanks!  Can we get a 2.3.2 release?

**Comment by doloopwhile** [comment]
@bobh66 
This is already fixed in 2.4.0.
